### PR TITLE
Remove some dead-code and duplication from o.e.transport

### PR DESCRIPTION
**Pull request by original-brownbear**
Removing some unused things and deduplicating connection lookup failure handling in TransportService.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/transport/TransportService.java [hunk 1]
```diff
@@ -750,21 +750,32 @@ public final <T extends TransportResponse> void sendRequest(
...
+    /**
+     * Get connection to the given {@code node} if possible and fail {@code handler} otherwise.
+     *
+     * @return connection if one could be found to {@code node} or null otherwise
+     */
+    @Nullable
+    private Transport.Connection getConnectionOrFail(DiscoveryNode node, String action, TransportResponseHandler<?> handler) {
         try {
-            connection = getConnection(node);
+            return getConnection(node);
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/transport/TransportService.java:767]: Paranoia: if some implementation of `getConnection()` itself returns `null` then we leak the handler here. Can we check for that case just to be sure?

Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/transport/TransportService.java]: makes sense, I pushed [9eec9a6](https://github.com/elastic/elasticsearch/pull/98321/commits/9eec9a6e70eeb05ef22dc5871a50bf17d1e1ea74)

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks David!

**Pull request merged**